packages/shared/createEventHook/index.ts

```diff
@@ -39,7 +39,7 @@ export function createEventHook<T = any>(): EventHook<T> {
   }
 
   const trigger: EventHookTrigger<T> = (param?: T) => {
-    return Promise.all(Array.from(fns).map(fn => param ? fn(param) : (fn as Callback<void>)()))
+    return Promise.all(Array.from(fns).map(fn => param !== undefined ? fn(param) : (fn as Callback<void>)()))
```

Comment:
If this is the case, we can ensure that the length of the `arguments` is correct.
```ts
  const trigger: EventHookTrigger<T> = (...args) => {
    return Promise.all(Array.from(fns).map(fn => fn(...(args as [T]))))
  }
```